library/psa_crypto.c

```diff
@@ -4665,6 +4684,38 @@ psa_status_t psa_key_agreement( psa_crypto_generator_t *generator,
     return( status );
 }
 
+psa_status_t psa_key_agreement_raw_shared_secret(psa_algorithm_t alg,
+                                                 psa_key_handle_t private_key,
+                                                 const uint8_t *peer_key,
+                                                 size_t peer_key_length,
+                                                 uint8_t *output,
+                                                 size_t output_size,
+                                                 size_t *output_length)
+{
+    psa_key_slot_t *slot;
+    psa_status_t status;
+
+    if( ! PSA_ALG_IS_KEY_AGREEMENT( alg ) )
+        return( PSA_ERROR_INVALID_ARGUMENT );
+    status = psa_get_key_from_slot( private_key, &slot,
+                                    PSA_KEY_USAGE_DERIVE, alg );
+    if( status != PSA_SUCCESS )
+        return( status );
+
+    status = psa_key_agreement_raw_internal( alg, slot,
+                                             peer_key, peer_key_length,
+                                             output, output_size,
+                                             output_length );
+    if( status != PSA_SUCCESS )
+    {
+        /* Fill the output buffer with random data, in case the caller
+         * ignores the error and attempts to use the result as a key
+         * to protect sensitive data. */
```

Comment:
Good idea. Since I was rebasing anyway I modified the original commit.

`psa_generator_import_key` may warrant similar protection, but that's out of scope of this PR.